### ISSUE TITLE
Unexpected behavior for quadratic terms

### ISSUE THREAD

**Issue by felixthoemmes** [rvlenth/emmeans#45]
Hello,
This may be an issue, or maybe it's a misunderstanding on my part, but I was surprised by the behavior of emmeans when a polynominal term of the form "I(x^2)" is in an lm statement. 

Using the I(x^2) notation, or creating a new squared term by hand (xsq <- x^2), and then using one or the other in an lm statement, and then feeding that into emmeans, yields different results. As far as I can tell inputting the already squared term by hand, yields the correct result, whereas the I(x^2) notation does not. 

Here is a reproducible example.
I generate a bunch of data, then I generate the squared term by hand, or insert it directly in the lm statement. I include all interactions in the lm model. 

The models are called lm.a and lm.b. Feeding them into emmeans, hoping to get the average treatment effect yields two different results. As far as I can tell the results of lm.b correspond with what would be called the average causal effect in the causal inference lit. 

`set.seed(1111)`
`preacademic <- rnorm(100,100,20)`
`p <- (exp(-2.3+.01*preacademic)) / (1/(1+exp(-2.3+.01*preacademic)))`
`treat <- rbinom(100,1,p)`
`postacademic<- 100 + 5*treat + 2*preacademic + rnorm(100,0,20)`
`treat <- factor(treat)`
`levels(treat) <- c("control","treatment")`
`df1 <- data.frame(preacademic,treat,postacademic)`
`df1$preacademic2 <- df1$preacademic^2`
`lm.a <-lm(postacademic~treat+preacademic+I(preacademic^2)+treat:preacademic+treat:I(preacademic^2))`
`lm.b <- lm(postacademic~treat+preacademic+preacademic2+treat:preacademic+treat:preacademic2)`
`summary(emmeans(lm.a,"treat",contr="pairwise",weights="proportinal"),infer=TRUE)`
`summary(emmeans(lm.b,"treat",contr="pairwise",weights="proportinal"),infer=TRUE)`

The last two commands produce the following output:

`$contrasts
 contrast             estimate       SE df  lower.CL upper.CL t.ratio p.value
 control - treatment -11.54449 5.067913 94 -21.60695 -1.48203  -2.278  0.0250

$contrasts
 contrast             estimate       SE df  lower.CL  upper.CL t.ratio p.value
 control - treatment -10.63247 4.133922 94 -18.84047 -2.424475  -2.572  0.0117
`
To compare my results, I generate predicted values for all individuals setting either all to treatment or all to control, and then taking a mean difference of those. The results are always the same, and always identical to the emmeans model with lm.b.

`#treatment effect by hand using predictions
mean(predict(lm.a,newdata = data.frame(treat=factor(rep("control",100))))) - mean(predict(lm.a,newdata = data.frame(treat=factor(rep("treatment",100)))))

#treatment effect by hand using predictions
mean(predict(lm.b,newdata = data.frame(treat=factor(rep("control",100))))) - mean(predict(lm.b,newdata = data.frame(treat=factor(rep("treatment",100)))))
`

Thanks!

**Comment by rvlenth** [comment]
The way it is done with `lm.a` is the correct one,. The reason is because of the way EMMs are defined in terms odf a reference grid (see https://cran.r-project.org/web/packages/emmeans/vignettes/basics.html#ref_grid). Note that, unless you specify otherwise, covariates are reduced to their means. Let's take a look at the reference grids for each of your models (BTW, I had to add `data = df1` to each `lm()` call to get these to work):

```
> emmeans::ref_grid(lm.a)
'emmGrid' object with variables:
    treat = control, treatment
    preacademic = 104.83

> emmeans::ref_grid(lm.b)
'emmGrid' object with variables:
    treat = control, treatment
    preacademic = 104.83
    preacademic2 = 11472
```
In both cases, we make predictions with `preacademic` set to 104.83. But in `lm.b`, we set `preacademic2` to 11472. But the problem here is that `preacademic2` is supposed to be the square of `preacademic`, and 104.83^2 = 10989, not 11472 (with is the mean of the square, rather than the square of the mean). It doesn't make sense to set a covariate to one value, and its square to something other than the actual square of the first.  In the computations for `lm.a`, the square is actually computed based on the value of `preacademic`:
```
> emmeans::ref_grid(lm.a)@linfct
  (Intercept) treattreatment preacademic I(preacademic^2) treattreatment:preacademic
1           1              0    104.8302         10989.36                     0.0000
2           1              1    104.8302         10989.36                   104.8302
  treattreatment:I(preacademic^2)
1                            0.00
2                        10989.36
```
In the old **lsmeans** package, I made mention of this in the user guide; but I seem to have omitted that in the assorted vignettes now provided in **emmeans**. I will fix that in the next update.

**Comment by felixthoemmes** [comment]
Thanks for the quick reply, Russ. It's entirely possible that I have overlooked this in the vignette. A quick search in both the lsmeans and emmeans vignette seems to confirm though, that it does not appear in the emmeans vignette anymore. 
I guess I have been using lsmeans / emmeans for such a long time now, that I don't use the help that often anymore - but I still find these little things that surprise me. In any event, thanks again! Closing this comment now.
